services/jobservices/src/condor-launcher/condor.go

```diff
@@ -0,0 +1,425 @@
+	if err != nil {
+		return "", "", "", err
+	}
+	cmdPath := path.Join(dir, "iplant.cmd")
+	configPath := path.Join(dir, "config")
+	jobPath := path.Join(dir, "job")
+	irodsPath := path.Join(dir, "irods-config")
+	err = ioutil.WriteFile(cmdPath, []byte(cmdContents), 0644)
+	if err != nil {
+		return "", "", "", nil
+	}
+	err = ioutil.WriteFile(configPath, []byte(jobConfigContents), 0644)
+	if err != nil {
+		return "", "", "", nil
+	}
+	err = ioutil.WriteFile(jobPath, []byte(jobContents), 0644)
+	if err != nil {
+		return "", "", "", nil
+	}
+	err = ioutil.WriteFile(irodsPath, []byte(irodsContents), 0644)
+	return cmdPath, configPath, jobPath, err
+}
+
+func submit(cmdPath string, s *model.Job) (string, error) {
+	csPath, err := exec.LookPath("condor_submit")
+	if err != nil {
+		return "", err
+	}
+	if !path.IsAbs(csPath) {
+		csPath, err = filepath.Abs(csPath)
+		if err != nil {
+			return "", err
+		}
+	}
+	cmd := exec.Command(csPath, cmdPath)
+	cmd.Dir = path.Dir(cmdPath)
+	pathEnv, err := configurate.C.String("condor.path_env_var")
+	if err != nil {
+		pathEnv = ""
+	}
+	condorCfg, err := configurate.C.String("condor.condor_config")
+	if err != nil {
+		condorCfg = ""
+	}
+	cmd.Env = []string{
+		fmt.Sprintf("PATH=%s", pathEnv),
+		fmt.Sprintf("CONDOR_CONFIG=%s", condorCfg),
+	}
+	output, err := cmd.CombinedOutput()
+	log.Printf("Output of condor_submit:\n%s\n", output)
+	if err != nil {
+		return "", err
+	}
+	log.Printf("Extracted ID: %s\n", string(model.ExtractJobID(output)))
+	return string(model.ExtractJobID(output)), err
+}
+
+func launch(s *model.Job) (string, error) {
+	sdir, err := CreateSubmissionDirectory(s)
+	if err != nil {
+		log.Printf("Error creating submission directory:\n%s\n", err)
+		return "", err
+	}
+	cmd, _, _, err := CreateSubmissionFiles(sdir, s)
+	if err != nil {
+		log.Printf("Error creating submission files:\n%s", err)
+		return "", err
+	}
+	id, err := submit(cmd, s)
+	if err != nil {
+		log.Printf("Error submitting job:\n%s", err)
+		return "", err
+	}
+	log.Printf("Condor job id is %s\n", id)
+	return id, err
+}
+
+func stop(s *model.Job) (string, error) {
+	crPath, err := exec.LookPath("condor_rm")
+	log.Printf("condor_rm found at %s", crPath)
+	if err != nil {
+		return "", err
+	}
+	if !path.IsAbs(crPath) {
+		crPath, err = filepath.Abs(crPath)
+		if err != nil {
+			return "", err
+		}
+	}
+	pathEnv, err := configurate.C.String("condor.path_env_var")
+	if err != nil {
+		pathEnv = ""
+	}
+	condorConfig, err := configurate.C.String("condor.condor_config")
+	if err != nil {
+		condorConfig = ""
+	}
+	cmd := exec.Command(crPath, s.CondorID)
+	cmd.Env = []string{
+		fmt.Sprintf("PATH=%s", pathEnv),
+		fmt.Sprintf("CONDOR_CONFIG=%s", condorConfig),
+	}
+	output, err := cmd.CombinedOutput()
+	log.Printf("condor_rm output for job %s:\n%s\n", s.CondorID, string(output))
+	if err != nil {
+		return "", err
+	}
+	return string(output), err
+}
+
+// AppVersion prints version information to stdout
+func AppVersion() {
```

Comment:
This functionality is duplicated a few times. Very minor, but it could be extracted. It seems that we may be having more `golang`-based projects in the future, so having a common lib which allows us to print app version statements in a consistent manner may be a good idea.

Not necessary for this PR. Would you like me to create an issue?